google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -243,7 +244,8 @@ def find_violations(self, instance_network_interface_list):
                     project=project,
                     network=network,
                     ip=ips,
-                    raw_data=repr(instance_network_interface))
+                    raw_data=json.dumps(instance_network_interface.__dict__,
```

Comment:
Why not store the original JSON inside of the instance_network_interface object as a private attribute with a getter property.

In common/gcp_type/instance.py, class InstanceNetworkInterface.__init__() set self._json = json.dumps(kwargs, sort_keys=True)

Then add a new method:

def as_json(self):
  return _json

Then you can set raw_data to instance_network_interface.as_json()

Note that an as_json method would be a nice base method for all gcp_type objects.

Reply:
Really nice suggestion, done!  Added a TODO for the base method.